scripts/o3de/o3de/project_properties.py

```diff
@@ -81,7 +79,14 @@ def edit_project_props(proj_path: pathlib.Path = None,
                        delete_gem_names: str or list = None,
                        replace_gem_names: str or list = None,
                        new_engine_name: str or list = None,
-                       is_optional: bool = False
+                       new_compatible_engines: str or list = None,
+                       delete_compatible_engines: str or list = None,
+                       replace_compatible_engines: str or list = None,
+                       new_version: str = None,
+                       is_optional: bool = False,
+                       new_engine_api_dependencies: list or str = None,
+                       delete_engine_api_dependencies: list or str = None,
+                       replace_engine_api_dependencies: list or str = None
```

Comment:
I wish we had pydoc for these arguments.

@amzn-phist  Such as the user doesn't know that the `is_optional` parameter is for setting optional gem dependencies into the "gem_names" field.
The name for the parameter `is_optional` doesn't really provide context without any documentation.
The user could read the method signature and ask the question "What is optional?"

Reply:
Thanks @AMZN-alexpete